src/query_ctx.c

```diff
@@ -93,8 +93,7 @@ void QueryCtx_SetError(char *error) {
 	QueryCtx *ctx = _QueryCtx_GetCtx();
 	// An error is already set.
 	if(ctx->internal_exec_ctx.error) free(error);
-
-	ctx->internal_exec_ctx.error = error;
+	else ctx->internal_exec_ctx.error = error;
```

Comment:
if an error is already set, either free it and set the new error, or drop the new error and keep the existing one,
placing this else here will free an existing one but it won't set the new error.